### ISSUE TITLE
Use ES6 module syntax

### ISSUE THREAD

**Issue by tanx** [openpgpjs/openpgpjs#385]
Hey,

I've been experimenting with ES6 and babel in hoodiecrow and am really liking ES6 modules. The main advantage is that you don't need a build step at development time since the browser can resolve the modules natively. It also works well with browserify using the babelify so that you can just consume common.js modules like before using the new syntax.

May proposal would be to refactor the following:

    var lib = require('lib');

to

    import lib from 'lib';

and

    module.exports = lib;

to

    export default lib;

It's really easy to integrate babel into the browserify grunt build. See my setup here:

https://github.com/tanx/hoodiecrow/blob/master/Gruntfile.js#L15-L19

The cool thing is developers can choose to activate experimental ES6 features in chrome to work without a build step during development:

https://github.com/nodejs/node-v0.x-archive/wiki/ES6-%28a.k.a.-Harmony%29-Features-Implemented-in-V8-and-Available-in-Node

But this is purely optional and devs can keep using the current browserify build workflow as well if they wish.

Publishing to npm works like so:

http://mammal.io/articles/using-es6-today/

Let me know what you think. It's not a lot of work to refactor from common.js to es6 but I wanted to get you input before doing it.

Thanks

**Comment by bartbutler** [comment]
Sounds good to me. I'm in favor of anything that makes development easier as long as, as it appears here, we aren't having to sacrifice any compatibility for non-ES6 browsers.

**Comment by tanx** [comment]
Babel is an ES6 to ES5 transpiler. All generated code can be executed in all runtimes that we currently serve today.

**Comment by bartbutler** [comment]
Very nice.

**Comment by tanx** [comment]
Sorry forgot the link: http://babeljs.io

**Comment by toberndo** [comment]
Good to embrace ES6. Is there a chance to leave out browserify and only use babel for the builds?

**Comment by tanx** [comment]
We could migrate completely from browserify to babel, but I would recommend that as a second step once we feel comfortable with the new syntax. Browserify works well for now and the transition will probably be smoother if we can still consume common.js modules, since I don't know if all of our dependencies expose a ES6 style module.

**Comment by evilaliv3** [comment]
:+1:

**Comment by tanx** [comment]
Also keep in mind that with browserify our build exposes a UMD module which allows users of OpenPGP.js to consume either with common.js, AMD, window.openpgp or ES6 styles.

**Comment by tanx** [comment]
> Is there a chance to leave out browserify and only use babel for the builds?

@toberndo I've tried using babel directly to build `src/index.js`. I found out some interesting stuff that I thought I'd share. It seems that babel just transpiles single files from:

    ES6/modules --> ES5/common.js

It does not seem to understand recursive module dependencies or try to resolve them. Hence the usage as a browserify transform in the form of babelify. Each files is transpiled separately and piped to browserify as it builds the complete dependency tree. Browserify then resolves the ES5/common.js code as usual to a single executable bundle or in our case standalone library. This is similar to the coffeescript development workflow with coffeeify.

So looks like we're stuck with browserify (at least until ES6 is widely supported in browsers and node.js). But that's probably ok, since it has become quite good over the last few years.
